### PR TITLE
Add state helper functions to modules

### PR DESCRIPTION
**Pull request by simensrostad**
- Align modules with 'equal' set of helper function that
   sets states and substates, aswell as handles printing of state
   transitions.

 - Remove prefix from state variables and enums in managers

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -63,10 +63,10 @@ enum app_state {
  * Passive mode: Sensor data and GPS position is acquired when movement is
  *		 detected, or after the configured movement timeout occurs.
  */
-enum app_sub_state {
-	APP_SUB_STATE_ACTIVE_MODE,
-	APP_SUB_STATE_PASSIVE_MODE,
-} app_sub_state;
+static enum sub_state_type {
+	STATE_ACTIVE_MODE,
```
Comment (jtguggedal) [inline src/main.c]: Prefix with `SUB_STATE` to align with variable name ?

File: src/main.c [hunk 2]
```diff
@@ -104,56 +104,56 @@ static struct module_data self = {
...
+		return "STATE_INIT";
+	case STATE_RUNNING:
+		return "STATE_RUNNING";
 	default:
 		return "Unknown";
 	}
 }
 
-static char *sub_state2str(enum app_sub_state state)
+static char *sub_state2str(enum sub_state_type state)
```
Comment (jtguggedal) [inline src/main.c]: `state` -> `new_state` to avoid confusion with global variable `state`

File: src/main.c (hunk 1 above)
Comment (simensrostad) [reply src/main.c]: Yep.

File: src/main.c (hunk 2 above)
Comment (simensrostad) [reply src/main.c]: Agreed. Updated the other modules as well.

**Review by jtguggedal — APPROVED** [review]

**Pull request merged**